packages/core/rendering/core-rendering-server-internal/src/views/styles.tsx

```diff
@@ -8,25 +8,44 @@
  */
 
 import React, { FC } from 'react';
-import type { DarkModeValue } from '@kbn/core-ui-settings-common';
+import { type DarkModeValue, ThemeName } from '@kbn/core-ui-settings-common';
 
 interface Props {
   darkMode: DarkModeValue;
+  themeName: ThemeName;
   stylesheetPaths: string[];
 }
 
-export const Styles: FC<Props> = ({ darkMode, stylesheetPaths }) => {
+export const Styles: FC<Props> = ({ darkMode, themeName, stylesheetPaths }) => {
   return (
     <>
-      {darkMode !== 'system' && <InlineStyles darkMode={darkMode} />}
+      {darkMode !== 'system' && <InlineStyles darkMode={darkMode} themeName={themeName} />}
       {stylesheetPaths.map((path) => (
         <link key={path} rel="stylesheet" type="text/css" href={path} />
       ))}
     </>
   );
 };
 
-const InlineStyles: FC<{ darkMode: boolean }> = ({ darkMode }) => {
+const InlineStyles: FC<{ darkMode: boolean; themeName: ThemeName }> = ({ darkMode, themeName }) => {
+  const getThemeStyles = (theme: ThemeName) => {
+    if (theme === 'borealis') {
+      return {
+        pageBackground: darkMode ? '#07101F' : '#F6F9FC', // colors.body
+        welcomeText: darkMode ? '#8E9FBC' : '#5A6D8C', // colors.subduedText
+        progress: darkMode ? '#172336' : '#ECF1F9', // colors.lightestShade
+        progressBefore: darkMode ? '#599DFF' : '#0B64DD', // colors.primary
+      };
+    }
+
+    return {
+      pageBackground: darkMode ? '#141519' : '#F8FAFD',
+      welcomeText: darkMode ? '#98A2B3' : '#69707D',
+      progress: darkMode ? '#25262E' : '#F5F7FA',
+      progressBefore: darkMode ? '#1BA9F5' : '#006DE4',
+    };
```

Comment:
Please forgive a potentially silly question on my part. Why do we need to use hardcoded color values here? Are these colors not already available in the EUI theme at this point?

Reply:
Afaik, no. These styles are not inside the EUI provider context in which they'd have access to the tokens. They sit higher/outside/earlier of it. It's the base (server) template that loads initially. (loading screen)